### PR TITLE
update envExec docs

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

update envExec docs

### CODE REVIEW

File: docs/templating_funcs.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ The `exec` function allows you to run a command, returning the stdout of the com
 {{ $cmdOutpot := exec "./mycmd" (list "arg1" "arg2" "--flag1") }}
 ```
 
+#### `envExec`
+The `envExec` function allows you to run a command with environment variables, returning the stdout of the command. When the command fails, the template rendering will fail with an error message.
+
+```yaml
+{{ $cmdOutpot := exec (dict "envKey" "envValue") "./mycmd" (list "arg1" "arg2" "--flag1") }}
```
Comment (mumoshu) [inline docs/templating_funcs.md:22]:
```suggestion
{{ $cmdOutpot := envExec (dict "envKey" "envValue") "./mycmd" (list "arg1" "arg2" "--flag1") }}
```

**Review by mumoshu — APPROVED** [review]
Thanks!
On a side note, how does `envExec` inherit the helmfile process's envvars? Does `mycmd` get executed within an environment has `envKey=envValue` in addition to the helmfile process env(hence the envvars are inherited), or does it get only `envKey=envValue` so other envvars like `PATH` isnt' available(hence not inherited)?

**Comment by yxxhero** [comment]
see: https://github.com/helmfile/helmfile/blob/7e2462ed62728457974b7fc41271e6f7f88db2e1/pkg/tmpl/context_funcs.go#L86-L91

**Comment by yxxhero** [comment]
@mumoshu

File: docs/templating_funcs.md [hunk 2]
```diff
@@ -15,6 +15,13 @@ The `exec` function allows you to run a command, returning the stdout of the com
 {{ $cmdOutpot := exec "./mycmd" (list "arg1" "arg2" "--flag1") }}
 ```
 
+#### `envExec`
+The `envExec` function allows you to run a command with environment variables, returning the stdout of the command. When the command fails, the template rendering will fail with an error message.
```
Comment (itscaro) [inline docs/templating_funcs.md:19]:
```suggestion
The `envExec` function allows you to run a command with environment variables declared on-the-fly in addition to existing environment variables, returning the stdout of the command. When the command fails, the template rendering will fail with an error message.
```

**Pull request merged**